src/main/kotlin/org/cascadebot/bot/rabbitmq/objects/GeneralReqResObjects.kt

```diff
@@ -0,0 +1,88 @@
+package org.cascadebot.bot.rabbitmq.objects
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import java.nio.charset.StandardCharsets
+
+
+/*
+    Status Codes
+
+    0   -  99   Success
+    100 - 199   Data Error
+    200 - 299   Server Error
+    300 - 399   Scripting Error
+    400 - 499   ?
+
+ */
+@Serializable
+@JvmInline
+value class StatusCode private constructor(val code: Int) {
+
+    companion object {
+
+        val Success = StatusCode(0)
+        val NotFound = StatusCode(100)
+        val BadRequest = StatusCode(101)
+
+        val ServerException = StatusCode(200)
+
+    }
+
+}
+
+@Serializable
+@JvmInline
+value class ErrorCode private constructor(val code: String) {
+
+    companion object {
+
+        val UserNotFound = ErrorCode("user_not_found")
+        val InvalidProperty = ErrorCode("invalid_property")
+        val InvalidMethod = ErrorCode("invalid_method")
+
+        val InvalidJsonFormat = ErrorCode("invalid_json")
+
+        fun fromException(e: Exception): ErrorCode {
+            return ErrorCode(e.javaClass.simpleName)
+        }
+
+    }
+
+
+}
```

Comment:
I have a feeling this enum is going to get big. Maybe not have a giant enum for this.

Reply:
We can do small enums split off by type of error if you want. So like with these codes we'd have an invalid category and a user category.

Reply:
What do you suggest?

Reply:
Have smaller objects that extends from error code that are separated out by category. So like in the above example we could have (obviously not exact, just an example)
```
class InvalidErrorCodes(): ErrorCode() {
    companion object {
        val InvalidProperty = InvalidErrorCodes("invalid_property")
        val InvalidMethod = InvalidErrorCodes("invalid_method")

        val InvalidJsonFormat = InvalidErrorCodes("invalid_json")
    }
}
```
```
class UserErrorCodes(): ErrorCode() {
    companion object {
        val UserNotFound = ErrorCode("user_not_found")
    }
}
```
Something like this, not exact obv, but just going through the list real quick (so not a complete list) of user errors for just custom commands we'd probably have have.
- User not found
- User not in guild
- Cannot set nick
- Cannot add role
- Cannot remove role
So i think keeping everything in one file is not practical considering that's just one object (and user is probably our smallest)